File: 1732-find-the-highest-altitude/solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    /**
     * Calculates the highest altitude reached by a biker.
     * We track the current altitude by accumulating gains and update
     * the maximum altitude observed so far.
     */
    int largestAltitude(std::vector<int>& gain) {
        int maxAltitude = 0;
        int currentAltitude = 0;

        for (int g : gain) {
            currentAltitude += g;
            // Update maxAltitude if the current path leads higher
            if (currentAltitude > maxAltitude) {
                maxAltitude = currentAltitude;
            }
        }

        return maxAltitude;
    }
};

File: 1732-find-the-highest-altitude/solution.py
from typing import List

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        """
        Calculates the highest altitude reached by a biker.
        Iterates through the gain array, updating the current altitude
        and tracking the peak value encountered.
        """
        max_altitude = 0
        current_altitude = 0
        
        for g in gain:
            current_altitude += g
            # Update max_altitude if the current point is higher
            if current_altitude > max_altitude:
                max_altitude = current_altitude
                
        return max_altitude